sources/api/netdog/src/networkd/devices/interface.rs

```diff
@@ -16,3 +19,32 @@ pub(crate) struct NetworkDInterface {
     pub(crate) static6: Option<StaticConfigV1>,
     pub(crate) routes: Option<Vec<RouteV1>>,
 }
+
+impl NetworkCreator for NetworkDInterface {
+    fn create_networks(&self, vlans: &Vlans) -> Vec<NetworkConfig> {
```

Comment:
:thinking:   That's probably a good callout.  

The only reason this is a bit odd, is because in net config, VLANs specify the device they're attached to.  But in networkd, the _device's_ config must contain the VLAN... 

I'm not sure if there is anything else like this, but to future proof I suppose passing in a struct makes sense.

Reply:
After poking around a bit more, I didn't find other situations where a devices config requires additional "things" from the overall net config.  I'm feeling more apt to leave this as-is until we find come into a situation where we need additional params. (something something YAGNI)

Reply:
This makes sense, I just wanted to make sure we weren't ignoring future work but it sounds like this might remain intact for the foreseeable future so I'm happy to leave it even if it does look a bit odd.